src/Tizen.System/PowerUsage/PowerUsageExceptionFactory.cs

```diff
@@ -0,0 +1,111 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using Tizen.Internals.Errors;
+
+namespace Tizen.System
+{
+    /// <summary>
+    /// Power Usage error codes.
+    /// </summary>
+    /// <since_tizen> 7 </since_tizen>
+    internal enum PowerUsageError
+    {
+        /// <summary>
+        /// Successful.
+        /// </summary>
+        None = ErrorCode.None,
+
+        /// <summary>
+        /// Out of memory error.
+        /// </summary>
+        OutOfMemory = ErrorCode.OutOfMemory,
+
+        /// <summary>
+        /// Invalid parameter.
+        /// </summary>
+        InvalidParameter = ErrorCode.InvalidParameter,
+
+        /// <summary>
+        /// No data available.
+        /// </summary>
+        NoData = ErrorCode.NoData,
+
+        /// <summary>
+        /// Permission denied.
+        /// </summary>
+        AcessibilityNotallowed = ErrorCode.PermissionDenied,
+
+        /// <summary>
+        /// Address family not supported.
+        /// </summary>
+        NotSupported = ErrorCode.NotSupported,
+
+        /// <summary>
+        /// Related record does not exist.
+        /// </summary>
+        RecordNotFound = -0x03060000 | 0x01,
+
+        /// <summary>
+        /// DB operation failed .
+        /// </summary>
+        DBFailed = -0x03060000 | 0x02,
+
+        /// <summary>
+        /// DB is not connected.
+        /// </summary>
+        DBNotOpened = -0x03060000 | 0x03,
+
+        /// <summary>
+        /// Internal error for generic use.
+        /// </summary>
+        Internal = -0x03060000 | 0x04
+    }
+
+    internal static class PowerUsageErrorFactory
+    {
+        internal const string LogTag = "Tizen.System.PowerUsage";
+        internal static Exception ThrowPowerUsageException(PowerUsageError errCode)
+        {
+            Log.Error(LogTag, "Throw PowerUsage Exception : " + errCode);
+
+            switch (errCode)
+            {
+                case PowerUsageError.OutOfMemory:
+                    return new InvalidOperationException("Out of memory");
+                case PowerUsageError.InvalidParameter:
+                    return new ArgumentException("Invalid Parameter passed");
+                case PowerUsageError.NoData:
+                    return new UnauthorizedAccessException("No data available");
+                case PowerUsageError.AcessibilityNotallowed:
+                    return new NotSupportedException("Permission denied");
+                case PowerUsageError.NotSupported:
+                    return new InvalidOperationException("Address family not supported");
+                case PowerUsageError.RecordNotFound:
```

Comment:
I think it is not exception case, We can return just 0 value for it

Reply:
It comes from native when the device does not support the PowerUsageResourceType.
Are you suggesting instead of throwing that exception, we should return 0 in that case ?

Reply:
Could you explain which C# API was affected by this exception?
We should control this error case by case

If `private static double GetPowerUsage(IntPtr dataHandle, PowerUsageResourceType rtype)` method throw InvalidOperationException, User hard to find reason of exception, because it used by `GetPowerUsage(string appID, IList<PowerUsageResourceType> rtypes, DateTime start, DateTime end)`
Exception does not know which ResourceType is not supported

So, if GetPowerUsage was throw exception, You need to catch it and ignore it like below
``` c#
foreach (PowerUsageResourceType type in rtypes)
{
    try
    {
        batteryUsage[type] = GetPowerUsage(dataHandle, type);
    }
    catch (InvalidOperationException)
    {
        Log.Error(PowerUsageErrorFactory.LogTag, $"Error getting battery usage for {type}");
    }
}
```

Reply:
PowerUsageError.RecordNotFound is thrown from following C# API's.
1)GetPowerUsage(PowerUsageResourceType rtype, DateTime start, DateTime end)
2)GetPowerUsage(string appID, PowerUsageResourceType rtype, DateTime start, DateTime end)
3)GetPowerUsage(IntPtr dataHandle, PowerUsageResourceType rtype)

You have given a very clear insight regarding how to handle it. Thank you.